### PR TITLE
fix server error when adding renamed github team as owner

### LINKED ISSUE #2687: Internal server error when trying to add GitHub team as owner after team or org rename
Here's what I'm trying to do:
```
cargo owner --add github:braun-embedded:developers
```

This results in the following error message:
```
error: failed to invite owners to crate dw1000: failed to get a 200 OK response, got 500
headers:
        HTTP/2 500 

        content-length: 21

        server: nginx

        date: Thu, 13 Aug 2020 11:11:12 GMT

        strict-transport-security: max-age=31536000

        via: 1.1 vegur, 1.1 e56e6732f380db727425bac2d6158761.cloudfront.net (CloudFront)

        x-cache: Error from cloudfront

        x-amz-cf-pop: FRA2-C2

        x-amz-cf-id: csEF2IVenu_CJy_wWXX6wTUXObJaG0P_12nmvLGcT2uz9g_F0rdGhA==



body:
Internal Server Error
```

This error message is from the `dw1000` crate, but I've also tried the same on `dwm1001`, `embedded-timeout-macros`, and `ieee802154` before I gave up. So I think the problem is related to my user or the team, not those crates.

I've double checked the org name, team name, read the relevant documentation, and couldn't find any mistakes on my part. Of course I can't rule out that I'm doing something wrong, but the error message is less than helpful :-)

### LINKED ISSUE #2957: `cargo owner -a` fails with HTTP 500 error
I'm attempting to add a GitHub team (@rustcrypto/ring-compat) as the owner of the `ring-compat` crate. However, when I do so, I get a 500 response:

```
$ cargo owner -a github:rustcrypto:ring-compat
    Updating crates.io index
error: failed to invite owners to crate ring-compat: failed to get a 200 OK response, got 500
headers:
	HTTP/2 500

	content-length: 21

	server: nginx

	date: Sun, 25 Oct 2020 14:18:20 GMT

	strict-transport-security: max-age=31536000

	via: 1.1 vegur, 1.1 c1caaceb6655a57ae014aef7bc8ec389.cloudfront.net (CloudFront)

	x-cache: Error from cloudfront

	x-amz-cf-pop: SFO20-C1

	x-amz-cf-id: KNK-qspNvMWxuBEG_JPL3M3wL_isGKicdD5jFrSLyx1m4iR66_bq5g==



body:
Internal Server Error
```

The team was recently renamed and I suspect it might have something to do with that.

### PR DESCRIPTION
**Pull request by syphar**
This should fix #2957 and fix #2687, as I understood these bugs.

I made the changes as I thought I understood the project design, but I'm happy to implement any changes that will get this to match standards and design. 

IMHO the change from checking `login` to checking `github_id` only is fine. We always get current data from github for the name that the user requested to add. And there we always get the id from github and can treat it as PK. 

The change to `crates_by_team_id_not_including_deleted_owners` is needed because the `github:crates-test-org:core`  now is not allowed to be created with a new, different ID, which it would get from `new_team`

### CODE REVIEW

File: src/tests/team.rs [hunk 1]
```diff
@@ -109,6 +109,42 @@ fn nonexistent_team() {
...
+    let user = app.db_new_user(mock_user_on_both_teams().gh_login);
+    let token = user.db_new_token("arbitrary token name");
+
+    app.db(|conn| {
+        use cargo_registry::schema::teams::dsl::*;
+
+        CrateBuilder::new("foo_renamed_team", user.as_model().id).expect_build(conn);
+
+        // create team with same ID and different name compared to http mock
+        // used for `add_renamed_owner`
```
Comment (JohnTitor) [inline src/tests/team.rs:125]:
```suggestion
        // used for `add_named_owner`
```

Did you mean?

**Comment by syphar** [comment]
Anything else that needs changing?

**Review by JohnTitor — APPROVED** [review]
LGTM but I think it's worth bringing up this at the next meeting since Pietro originally left the instruction and other members may be interested in it.

**Comment by jtgeibel** [comment]
Originally the test setup confused me a bit with the call to `mock_user_on_both_teams`. I've pushed a new commit simplifying the setup a bit. Sorry for the delay in reviewing. I plan to deploy this fix later tonight.

@bors r+

**Pull request merged**